lib/topological_inventory/providers/common/collector.rb

```diff
@@ -30,7 +33,7 @@ def collect!
 
             targeted_refresh(notices) unless notices.empty?
 
-            sleep(poll_time)
+            standalone_mode ? sleep(poll_time) : stop
```

Comment:
Its a little strange that we have this here and also in the amazon and azure PRs.  Does every provider just overwrite the `collect!` method?

Maybe we split up collect_loop and collect and let the providers implement collect.  The collect method here doesn't look very reusable anyway

Reply:
yes, it should be refactored because I extracted it when we had only 2 providers, so I didn't know.

I agree with you but I would rather change it in another PR